sdk/testing/src/main/java/io/opentelemetry/sdk/testing/trace/TestSpanData.java

```diff
@@ -48,195 +96,454 @@ public static Builder builder() {
+      return this.name.equals(that.getName())
+          && this.kind.equals(that.getKind())
+          && this.spanContext.equals(that.getSpanContext())
+          && this.parentSpanContext.equals(that.getParentSpanContext())
+          && this.status.equals(that.getStatus())
+          && this.startEpochNanos == that.getStartEpochNanos()
+          && this.attributes.equals(that.getAttributes())
+          && this.events.equals(that.getEvents())
+          && this.links.equals(that.getLinks())
+          && this.endEpochNanos == that.getEndEpochNanos()
+          && this.totalRecordedEvents == that.getTotalRecordedEvents()
+          && this.totalRecordedLinks == that.getTotalRecordedLinks()
+          && this.totalAttributeCount == that.getTotalAttributeCount()
+          && this.resource.equals(that.getResource())
+          && this.internalHasEnded == that.hasEnded()
+          && this.instrumentationScopeInfo.equals(that.getInstrumentationScopeInfo());
+    }
+    return false;
+  }
 
-    abstract TestSpanData autoBuild();
+  @Override
+  public int hashCode() {
+    int hash = 1;
+    hash *= 1000003;
+    hash ^= name.hashCode();
+    hash *= 1000003;
+    hash ^= kind.hashCode();
+    hash *= 1000003;
+    hash ^= spanContext.hashCode();
+    hash *= 1000003;
+    hash ^= parentSpanContext.hashCode();
+    hash *= 1000003;
+    hash ^= status.hashCode();
+    hash *= 1000003;
+    hash ^= (int) ((startEpochNanos >>> 32) ^ startEpochNanos);
+    hash *= 1000003;
+    hash ^= attributes.hashCode();
+    hash *= 1000003;
+    hash ^= events.hashCode();
+    hash *= 1000003;
+    hash ^= links.hashCode();
+    hash *= 1000003;
+    hash ^= (int) ((endEpochNanos >>> 32) ^ endEpochNanos);
+    hash *= 1000003;
+    hash ^= totalRecordedEvents;
+    hash *= 1000003;
+    hash ^= totalRecordedLinks;
+    hash *= 1000003;
+    hash ^= totalAttributeCount;
+    hash *= 1000003;
+    hash ^= resource.hashCode();
+    hash *= 1000003;
+    hash ^= internalHasEnded ? 1231 : 1237;
+    hash *= 1000003;
+    hash ^= instrumentationScopeInfo.hashCode();
+    return hash;
+  }
 
-    abstract List<EventData> getEvents();
+  public static class Builder {
+    @Nullable private String name;
+    @Nullable private SpanKind kind;
+    @Nullable private SpanContext spanContext;
+    @Nullable private SpanContext parentSpanContext;
+    @Nullable private StatusData status;
+    @Nullable private Long startEpochNanos;
+    @Nullable private Attributes attributes;
+    @Nullable private List<EventData> events;
+    @Nullable private List<LinkData> links;
+    @Nullable private Long endEpochNanos;
+    @Nullable private Integer totalRecordedEvents;
+    @Nullable private Integer totalRecordedLinks;
+    @Nullable private Integer totalAttributeCount;
+    @Nullable private Resource resource;
+    @Nullable private Boolean internalHasEnded;
+    @Nullable private InstrumentationScopeInfo instrumentationScopeInfo;
+
+    public Builder() {}
```

Comment:
Discovered another instance of where we inadvertently exposed a default public constructor on an autovalue class. Needed to make this constructor public to maintain compatibility. Sigh..